lib/movable_ink/aws/ec2.rb

```diff
@@ -106,6 +107,64 @@ def instances(role:, exclude_roles: [], region: my_region, availability_zone: ni
         end
       end
 
+      def instances_with_consul_discovery(role:, region: my_region, availability_zone: nil)
+        if role == nil || role == ''
+          raise MovableInk::AWS::Errors::RoleNameRequiredError
+        end
+
+        if role.include? '_'
+          raise MovableInk::AWS::Errors::RoleNameInvalidError
+        end
+
+        Diplomat.configure do |config|
+          config.url = "https://localhost:8501"
+          config.options = { ssl: { verify: false } }
+        end
+
+        consul_instances = Diplomat::Service.get(role, :all, { :dc => datacenter(region) }).map { |node|
```

Comment:
Can you double check that this is using cached queries to consul?

Reply:
I went ahead and added `stale` and `cached` to all service queries.